core/auth/apis.go

```diff
@@ -611,6 +611,113 @@ func (a *Auth) CreateAdminAccount(authenticationType string, appID string, orgID
+		if err != nil {
+			return errors.WrapErrorAction(logutils.ActionFind, model.TypeAccount, nil, err)
+		}
+		if account == nil {
+			return errors.ErrorData(logutils.StatusMissing, model.TypeAccount, &logutils.FieldArgs{"app_org_id": appOrg.ID, "auth_type": authType.Code, "identifier": identifier})
+		}
+
+		//2. check if the user's auth type is verified
+		accountAuthType := account.GetAccountAuthType(authType.ID, identifier)
+		if accountAuthType == nil || accountAuthType.Unverified {
+			return errors.ErrorData("Unverified", model.TypeAccountAuthType, &logutils.FieldArgs{"app_org_id": appOrg.ID, "auth_type": authType.Code, "identifier": identifier}).SetStatus(utils.ErrorStatusUnverified)
+		}
+
+		//3. update account permissions
+		updatedAccount = account
+		updated := false
+		if account.CheckForPermissionChanges(permissions) {
+			newPermissions, err := a.CheckPermissions(context, appOrg, permissions, updaterPermissions)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionValidate, model.TypePermission, nil, err)
+			}
+			err = a.storage.UpdateAccountPermissions(context, account.ID, newPermissions)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionUpdate, "admin account permissions", nil, err)
+			}
+
+			updatedAccount.Permissions = newPermissions
+			updated = true
+		}
+
+		//4. update account roles
+		if account.CheckForRoleChanges(roleIDs) {
+			newRoles, err := a.CheckRoles(context, appOrg, roleIDs, updaterPermissions)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionValidate, model.TypeAppOrgRole, nil, err)
+			}
+
+			newAccountRoles := model.AccountRolesFromAppOrgRoles(newRoles, true, true)
+			err = a.storage.UpdateAccountRoles(context, account.ID, newAccountRoles)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionUpdate, "admin account roles", nil, err)
+			}
+
+			updatedAccount.Roles = newAccountRoles
+			updated = true
+		}
+
+		//5. update account groups
+		if account.CheckForGroupChanges(groupIDs) {
+			newGroups, err := a.checkGroups(context, *appOrg, groupIDs, updaterPermissions)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionValidate, model.TypeAppOrgGroup, nil, err)
+			}
+
+			newAccountGroups := model.AccountGroupsFromAppOrgGroups(newGroups, true, true)
+			err = a.storage.UpdateAccountGroups(context, account.ID, newAccountGroups)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionUpdate, "admin account groups", nil, err)
+			}
+
+			updatedAccount.Groups = newAccountGroups
+			updated = true
+		}
+
+		//6. delete active login sessions if account was updated
+		if updated {
+			err = a.storage.DeleteLoginSessionsByIdentifier(context, account.ID)
+			if err != nil {
+				return errors.WrapErrorAction(logutils.ActionDelete, model.TypeLoginSession, nil, err)
+			}
+
+			now := time.Now().UTC()
+			updatedAccount.DateUpdated = &now
+		}
```

Comment:
Hi @petyos, I think this was originally your suggestion in #411. Do you still think this is necessary, and if so, why? Thanks!

Reply:
Hi @shurwit , revoking a permission should trigger removing the login session because we have a window of time from the revoke moment and the token expiration where the token could access a resources which were revoked.

Yes, that time window is short and we should determine if this issue is a problem.